tests/integration/exec/server_suite_test.go

```diff
@@ -0,0 +1,197 @@
+				CpuMillis:   2000,
+				MemoryBytes: 2147483648,
+			},
+		},
+	}
+	machineClassData, err := json.Marshal(machineClasses)
+	Expect(err).NotTo(HaveOccurred())
+	machineClassesFile, err = os.CreateTemp(GinkgoT().TempDir(), "machineclasses")
+	Expect(err).NotTo(HaveOccurred())
+	Expect(os.WriteFile(machineClassesFile.Name(), machineClassData, 0600)).To(Succeed())
+	DeferCleanup(machineClassesFile.Close)
+	DeferCleanup(os.Remove, machineClassesFile.Name())
+
+	pluginOpts := networkinterfaceplugin.NewDefaultOptions()
+	pluginOpts.PluginName = "isolated"
+
+	// tempDir = "/home/pcmil/libvirt-provider/temptest"
+	tempDir = GinkgoT().TempDir()
+	Expect(os.Chmod(tempDir, 0730)).Should(Succeed())
+
+	opts := app.Options{
+		Address:                     filepath.Join(tempDir, "test.sock"),
+		BaseURL:                     baseURL,
+		PathSupportedMachineClasses: machineClassesFile.Name(),
+		RootDir:                     filepath.Join(tempDir, "libvirt-provider"),
+		StreamingAddress:            streamingAddress,
+		Servers: app.ServersOptions{
+			Metrics: app.HTTPServerOptions{
+				Addr: metricsAddress,
+			},
+			HealthCheck: app.HTTPServerOptions{
+				Addr: healthCheckAddress,
+			},
+		},
+		Libvirt: app.LibvirtOptions{
+			Socket:                "/var/run/libvirt/libvirt-sock",
+			URI:                   "qemu:///system",
+			PreferredDomainTypes:  []string{"kvm", "qemu"},
+			PreferredMachineTypes: []string{"pc-q35", "pc-i440fx"},
+			Qcow2Type:             "exec",
+		},
+		NicPlugin: pluginOpts,
+		// GCVMGracefulShutdownTimeout:    gracefulShutdownTimeout,
+		ResyncIntervalGarbageCollector: resyncGarbageCollectorInterval,
+		ResyncIntervalVolumeSize:       resyncVolumeSizeInterval,
+		GuestAgent:                     app.GuestAgentOption(api.GuestAgentNone),
+		MachineEventStore: machineevent.EventStoreOptions{
+			MachineEventMaxEvents:      5,
+			MachineEventTTL:            10 * time.Second,
+			MachineEventResyncInterval: 10 * time.Second,
+		},
+	}
+
+	srvCtx, cancel := context.WithCancel(context.Background())
+	DeferCleanup(cancel)
+
+	go func() {
+		defer GinkgoRecover()
+		Expect(app.Run(srvCtx, opts)).To(Succeed())
+	}()
+
+	Eventually(func() error {
+		return isSocketAvailable(opts.Address)
+	}).WithTimeout(30 * time.Second).WithPolling(500 * time.Millisecond).Should(Succeed())
+
+	address, err := machine.GetAddressWithTimeout(3*time.Second, fmt.Sprintf("unix://%s", opts.Address))
+	Expect(err).NotTo(HaveOccurred())
+
+	gconn, err := grpc.NewClient(address, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	Expect(err).NotTo(HaveOccurred())
+	DeferCleanup(gconn.Close)
+
+	machineClient = iriv1alpha1.NewMachineRuntimeClient(gconn)
+
+	c := dialers.NewLocal()
+	libvirtConn = libvirt.NewWithDialer(c)
+	Expect(libvirtConn.Connect()).To(Succeed())
+	Expect(libvirtConn.IsConnected(), BeTrue())
+	DeferCleanup(libvirtConn.ConnectClose)
+})
+
+func isSocketAvailable(socketPath string) error {
```

Comment:
this function is defined multiple times. Probably whole server_suite_test.go were copied. It has to be refactor.